great_expectations/util.py

```diff
@@ -888,6 +888,67 @@ def read_sas(
         )
 
 
+def build_in_memory_runtime_context() -> "BaseDataContext":  # noqa: F821
+    """
+    Create generic in-memory "BaseDataContext" context for manipulations as required by tests.
+    """
+    from great_expectations.data_context.data_context.base_data_context import (
+        BaseDataContext,
+    )
+    from great_expectations.data_context.types.base import (
+        DataContextConfig,
+        InMemoryStoreBackendDefaults,
+    )
+
+    data_context_config: DataContextConfig = DataContextConfig(
+        datasources={  # type: ignore[arg-type]
+            "pandas_datasource": {
+                "execution_engine": {
+                    "class_name": "PandasExecutionEngine",
+                    "module_name": "great_expectations.execution_engine",
+                },
+                "class_name": "Datasource",
+                "module_name": "great_expectations.datasource",
+                "data_connectors": {
+                    "runtime_data_connector": {
+                        "class_name": "RuntimeDataConnector",
+                        "batch_identifiers": [
+                            "id_key_0",
+                            "id_key_1",
+                        ],
+                    }
+                },
+            },
```

Comment:
Instead of a dictionary, could this be a `Dict[str, DatasourceConfig]`? I'd like to use rich types as much as possible.

Reply:
@cdkini I simply moved this from another module to avoid circular imports.  How about we attempt this in a follow-on effort?

Reply:
that works for me, thanks for the clarification